### DISCUSSION TITLE
Can I include an unescaped tilde in a Cask Formula `system_command`?

### DISCUSSION THREAD

**Discussion by unitof** [Homebrew/discussions#762] [Casks] [answered]
I'm trying to add a `system_command` block with `args` to a cask formula, and I need one of the `args` to be a subdirectory of the current user's home directory.

However, the `~` gets escaped when the formula is run, leading to this error after `brew install`:

```
Error: Failure while executing; `xattr -cr \~/Library/QuickLook/QLMarkdown.qlgenerator` exited with 1. Here's the output:
xattr: No such file: ~/Library/QuickLook/QLMarkdown.qlgenerator
```

It seems only [`#{appdir}`](https://github.com/Homebrew/brew/blob/4695220f56c7e621d1936066eade42106ef0457f/Library/Homebrew/cask/dsl.rb#L328-L330) is available as an expandable variable within Cask formula strings, unless there's some `#{homedir}` or similar available I'm missing.

If relevant (and open to anyone telling me this is a terrible idea), I'm trying to fix `qlmarkdown` Cask [refusing to run on anything post-Catalina](https://github.com/toland/qlmarkdown/issues/98#issuecomment-612259027) (even after being Allowed Anyway" in Security & Privacy prefpane, at least for me) by adding the following block to `qlmarkdown.rb`:

```
  postflight do
    system_command "xattr",
                   args: ["-cr", "~/Library/QuickLook/QLMarkdown.qlgenerator"],
                   sudo: false
  end
```

Thanks & love!

**Comment by gromgit** [comment, marked as answer]
Does:
```
postflight do
  system_command "xattr",
                 args: ["-cr", File.join(ENV["HOME"], "Library/QuickLook/QLMarkdown.qlgenerator")],
                 sudo: false
end
```
work?

**Reply by unitof** [reply]
Indeed it does! Nice move. Thanks so much.
